Django 1.5 - DO NOT MERGE

ksurya opened the pull request:
The below are the  changes made (some minor changes might have been missed but can be found clearly in commit messages)
1. Upgrade `{% url %}` template tag
2. Use django-haystack 2.1.0, xapian-haystack 2.0.0, Whoosh 2.5.1 -- make respective changes in `search_settings.py` and `search_indexes.py`
3. Use django-registration 1.0 -- Also move `registration.backends.default.urls` from `scipy_central.urls` to `scipy_central.person.urls`
4. DVCS (windows specific) error fixed
5. Use django transaction middleware, clickjacking middleware
6. Make `pages.views.page_404_error` default 404 handler
7. Add wsgi.py, change manage.py, upgrade `slugify` import, upgrade url conf import
8. Fix os.path.normpath import
9. Use paginator module for finding "next" or "prev" objects in Revision, Submission module
10. Make UserProfile model compatible with django-1.5

[5] Django transaction middleware raises https://github.com/scipy/SciPyCentral/issues/132

Note: A deprecation warning is observed when starting the server regarding `hashlib` import. This is caused in django transactions middleware (It will be deprecated in 1.7. However, we require it at the moment until we have our own Middleware)

ksurya committed:
ENH: filters added to mail_admins in logging

http://django.readthedocs.org/en/latest/releases/1.4.html#request-exceptions-are-now-always-logged

ksurya committed:
ENH: change url template tag to new-style

{% url view_name %} --> {% url 'view_name' %}

https://docs.djangoproject.com/en/1.5/releases/1.5/ (Overview Notes)

ksurya committed:
ENH: upgrade django-haystack

This commit requires following
1. django-haystack==2.1.0
2. xapian-haystack==2.0.0
3. xapian-python-bindings-for-python-2.7.0==1.2.8
4. Whoosh==2.5.1

This code is NOT backward compatible with older versions of
django-haystack, xapian-haystack
Kindly install xapian-haystack from its repository and NOT with `pip`
https://github.com/notanumber/xapian-haystack/ (master)

ksurya committed:
ENH: upgrade django-registration

Changes:

1. move built-in registration.backends.default.urls into person.urls (from
scipy_central.urls to scipy_central.person.urls)

2. django-registration latest version NOT compatible with previous
versions. Its majorly written from scratch using class-based `views`
(registration.backends.default.views.RegistrationView)

3. SciPyRegistrationBackend() is inherited from class-based view [2] and
moved into views.py

required packages:
django-registration==1.0

ksurya committed:
ENH: upgrade url conf imports

django.conf.urls.defaults import * --> django.conf.urls import *

ksurya committed:
ENH: use execute_from_command_line in manage.py

`execute_manager` is deprecated -- mentioned in django-1.4 release notes
update manage.py according to django-1.5

ksurya committed:
ENH: remove admin_media_prefix settings

admin_media_prefix is deprecated in django-1.4
Its used from static_url

ksurya committed:
ENH: make UserProfile compatible with django 1.5

Django 1.5 deprecates AUTH_PROFILE_MODULE settings which enable to extend User model.

scipy central UserProfile fields are not used in User authentication or Registration! So, a OneToField with User module fits our requirement. It is described at https://docs.djangoproject.com/en/1.5/topics/auth/customizing/#extending-the-existing-user-model

The following are the changes done:

1. remove AUTH_PROFILE_MODULE variable in settings
2. Improve UserProfile admin interface
3. change `User` model import

lomegor commented on deploy/wsgi.py:
```diff
@@ -0,0 +1,32 @@
+might make sense to replace the whole Django WSGI application with a custom one
+that later delegates to the Django one. For example, you could introduce WSGI
+middleware here, or combine a Django application with an application of another
+framework.
```
Remove this previous paragraph, it's not needed.

ksurya committed:
ENH: add wsgi application

1. deploy/wsgi.py

ksurya committed:
ENH: update slugify() import

from django.template.defaultfilters import slugify -->
from django.utils.text import slugify

change string type to unicode in tests
